src/main/mule/sources/anypoint/platform/apis/api-call-api-platform.xml

```diff
@@ -28,6 +29,61 @@ http://www.mulesoft.org/schema/mule/ee/core http://www.mulesoft.org/schema/mule/
 	"targetAdminSite" : p('anypoint.platform.apis.apiPlatform.clients.targetAdminSite')
 }]]]></http:query-params>
 		</http:request>
+		<ee:transform doc:name="Get number of API Clients Pages" doc:id="22c75968-ad73-48dc-a94e-3f41253842e0" >
+			<ee:message >
+			</ee:message>
+			<ee:variables >
+				<ee:set-variable variableName="apiClientsPages" ><![CDATA[%dw 2.0
+output application/java
+var pageLimit = (Mule::p('anypoint.platform.apis.apiPlatform.clients.limit') as Number default 100)
+---
+((1 to ceil((vars.apiClientsResponsePayload."total" default 0)/pageLimit)) - 1 default [])]]></ee:set-variable>
+			</ee:variables>
+		</ee:transform>
+		<logger level="INFO" doc:name="Logger" doc:id="4fe0e979-f9be-46be-b475-97cee57a21e1" message="limit #[Mule::p('anypoint.platform.apis.apiPlatform.clients.limit')] offset p('anypoint.platform.apis.apiPlatform.clients.offset')"/>
+		<foreach doc:name="For Each" doc:id="0268328b-51dc-4934-83e6-174f243d52cf" collection="#[vars.apiClientsPages]">
+			<http:request method="GET" doc:name="Additional API Clients Request" doc:id="e1b11283-7ae6-4aa7-842f-b08fe8c2f07e" config-ref="HTTP_Request_configuration" path="${anypoint.platform.apis.apiPlatform.clients.path}" >
+				<http:body ><![CDATA[#[null]]]></http:body>
+				<http:headers ><![CDATA[#[output application/java
+---
+{
+	"Authorization" : "Bearer " ++ vars.token
+}]]]></http:headers>
+				<http:uri-params ><![CDATA[#[output application/java
+---
+{
+	"orgId" : vars.orgId
+}]]]></http:uri-params>
+				<http:query-params ><![CDATA[#[output application/java
+---
+{
+	"offset" : vars.counter * (Mule::p('anypoint.platform.apis.apiPlatform.clients.limit') as Number) ,
+	"limit" : Mule::p('anypoint.platform.apis.apiPlatform.clients.limit'),
+	"targetAdminSite" : Mule::p('anypoint.platform.apis.apiPlatform.clients.targetAdminSite')
+}]]]></http:query-params>
+			</http:request>
+			<ee:transform doc:name="Aggregate API Clients Results" doc:id="2b84ae90-47d8-467a-9924-9413c01df060" >
+				<ee:message >
+				</ee:message>
+				<ee:variables >
+					<ee:set-variable variableName="apiClientsResponsePayload" ><![CDATA[%dw 2.0
+output application/java
+---
+{
+	applications: (payload.applications ++ vars.apiClientsResponsePayload.applications default null),
+	total: payload.total default null
```

Comment:
Hi @roltra , I don't see total being used after the aggregation, is it needed to map the value?

Reply:
I don't believe it is being used, I just left as is in case someone was using through the raw call - since it will return the original structure of the call, if that makes sense

Reply:
I understand, it tries to maintain the structure of the original response coming from the API, only extending the applications var